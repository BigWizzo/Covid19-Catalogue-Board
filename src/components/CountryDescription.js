import React from 'react';
import PropTypes from 'prop-types';

const CountryDescription = ({ description }) => (
  <div className="mb-3">
    <div className="row g-0">
      <div className="col-md-8">
        <div className="card-body text-left">
          <p className="country-wrap font-22">
            <span>Country Name: </span>
            <span className="country-code green-800 font-22">
              {description?.label.toUpperCase()}
            </span>
          </p>
          <p className="card-text">
            <span>Country Code: </span>
            <span className="card-title">{description?.abbreviation}</span>
          </p>
          <p className="mute-p">
            <p className="text-muted">
              {description?.label} has a population of {description?.population}
              . It has so far recorded {description?.deaths} deaths.{' '}
              {description?.confirmed} cases have been confirmed and{' '}
              {description?.recovered} people have recovered from the virus.
            </p>
          </p>
        </div>
      </div>
      <div className="col-md-4 text-center">
        <h4 className="card-hed font-italic">{description?.continent}</h4>
        <hr />
        <div className="card-body">
          <div className="icon-div py-3">
            <div className="icon-span ">
              <div className="">
                <p className="font-10 text-success">Last Updated</p>
                <p className="increase">
                  <i className="fas fa-pen-square font-22 text-success" />
                </p>
                <p className="font-12 text-success">{description?.updated}</p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
);

CountryDescription.propTypes = {
  description: PropTypes.string.isRequired,
};

export default CountryDescription;
