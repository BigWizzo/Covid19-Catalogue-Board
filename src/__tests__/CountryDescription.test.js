import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store, persistor } from '../redux/store';
import { PersistGate } from 'redux-persist/integration/react';
import CountryDescription from '../components/CountryDescription';

test('should render the country description component', () => {
  const component = renderer.create(
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <BrowserRouter>
          <CountryDescription />
        </BrowserRouter>
      </PersistGate>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
