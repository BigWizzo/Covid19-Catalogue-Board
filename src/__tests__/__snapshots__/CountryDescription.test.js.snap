// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the country description component 1`] = `
<div
  className="mb-3"
>
  <div
    className="row g-0"
  >
    <div
      className="col-md-8"
    >
      <div
        className="card-body text-left"
      >
        <p
          className="country-wrap font-22"
        >
          <span>
            Country Name: 
          </span>
          <span
            className="country-code green-800 font-22"
          />
        </p>
        <p
          className="card-text"
        >
          <span>
            Country Code: 
          </span>
          <span
            className="card-title"
          />
        </p>
        <p
          className="mute-p"
        >
          <small
            className="text-muted"
          >
             has a population of
             
            . It has so far recorded
             
             deaths. 
             cases have been confirmed and 
             people have recovered from the virus.
          </small>
        </p>
      </div>
    </div>
    <div
      className="col-md-4 text-center"
    >
      <h4
        className="card-hed font-italic"
      />
      <hr />
      <div
        className="card-body"
      >
        <div
          className="icon-div py-3"
        >
          <div
            className="icon-span "
          >
            <div
              className=""
            >
              <p
                className="font-10 text-success"
              >
                Last Updated
              </p>
              <p
                className="increase"
              >
                <i
                  className="fas fa-pen-square font-22 text-success"
                />
              </p>
              <p
                className="font-12 text-success"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
